ecs-agent/logger/log.go

```diff
@@ -144,25 +160,35 @@ func reloadConfig() {
 }
 
 func seelogConfig() string {
+	platformLogConfig := platformLogConfig()
 	c := `
 <seelog type="asyncloop">
-	<outputs formatid="` + Config.outputFormat + `">
+	<outputs formatid="` + Config.outputFormat + `">`
+	if platformLogConfig != "" || Config.logToStdout {
+		c += `
 		<filter levels="` + getLevelList(Config.driverLevel) + `">
-			<console />`
-	c += platformLogConfig()
-	c += `
+		`
+		if Config.logToStdout {
+			c += `	<console />`
+		}
+		c += platformLogConfig
+		c += `
 		</filter>`
+	}
 	if Config.logfile != "" {
 		c += `
 		<filter levels="` + getLevelList(Config.instanceLevel) + `">`
 		if Config.RolloverType == "size" {
 			c += `
 			<rollingfile filename="` + Config.logfile + `" type="size"
 			 maxsize="` + strconv.Itoa(int(Config.MaxFileSizeMB*1000000)) + `" archivetype="none" maxrolls="` + strconv.Itoa(Config.MaxRollCount) + `" />`
-		} else {
+		} else if Config.RolloverType == "date" {
```

Comment:
This seems like a behavior change as before not setting `RolloverType` would default to date based rollover but not it won't. Is this intended?

Reply:
`DEFAULT_ROLLOVER_TYPE` is set to date. I can definitely switch this `if` and `else if` to speed up the access time to the date setting.

Reply:
I think it's a behavior change for the logger as currently the logger defaults to date based rolling if the rollover type is not "size". I'd prefer that we keep the behavior of the logger same as before if this PR is about standardizing the configuration methods for the logger.

Reply:
synced offline, we can ensure no behavioral change by keeping "date" with a rollingfile config as the default and checking for 'none' to use a simple log file.